homeassistant/components/rainmachine/__init__.py

```diff
@@ -320,6 +341,17 @@ async def async_pause_watering(call: ServiceCall, controller: Controller) -> Non
         """Pause watering for a set number of seconds."""
         await controller.watering.pause_all(call.data[CONF_SECONDS])
 
+    @call_with_controller(update_programs_and_zones=False)
+    async def async_push_flow_meter_data(
+        call: ServiceCall, controller: Controller
+    ) -> None:
+        """Push flow meter data to the device."""
+        value = call.data[CONF_VALUE]
+        if units := call.data.get(CONF_UNIT_OF_MEASUREMENT):
+            await controller.watering.post_flowmeter(value=value, units=units)
+            return
```

Comment:
Even though it's functionally the same, a quick return here feels odd – not really guarding anything. I would just use a straight if/else.

Reply:
Change made. Blame bdraco for beating quick returns into me previously.